packages/wallet/ui/src/components/SmartWalletConnection.jsx

```diff
@@ -45,11 +54,32 @@ const SmartWalletConnection = ({
       message += `: ${e.message}`;
     }
     if (severity === 'error') {
-      setConnectionState('error');
+      setConnectionStatus(ConnectionStatus.Error);
     }
     setSnackbarMessages(sm => [...sm, { severity, message }]);
   };
 
+  const { href, smartConnectionMethod } = connectionConfig;
+  let publicAddress;
+  if (
+    smartConnectionMethod === SmartConnectionMethod.KEPLR &&
+    keplrConnection
+  ) {
+    publicAddress = keplrConnection.address;
+  } else if (smartConnectionMethod === SmartConnectionMethod.READ_ONLY) {
+    publicAddress = connectionConfig.publicAddress;
+  }
```

Comment:
consider a helper method so this is one line here assigning to a const

Reply:
Done